src/main/kotlin/calculator/ExpressionParser.kt

```diff
@@ -0,0 +1,12 @@
+package calculator
+
+class ExpressionParser {
+    companion object {
+        fun <T> Iterable<String>.parse(filterByIndex: (Int) -> Boolean, convertByType: (String) -> T): List<T> {
```

Comment:
아래와 같이 체이닝으로 표현되는 경우 아래와 같이 표현도 가능합니다.

```suggestion
        fun <T> Iterable<String>.parse(filterByIndex: (Int) -> Boolean, convertByType: (String) -> T): List<T> =
```